src/filehandler/ifile.cpp

```diff
@@ -1,36 +1,55 @@
 #include "filehandler/ifile.h"
+#include "common.h"
+
 #include <fstream>
 #include <utility>
 
-IFile::IFile(std::string path) :
-        m_filePath(std::move(path))
-{}
+IFile::IFile(std::string path, std::string expectedExtension) :
+        m_filePath(std::move(path)),
+        m_expectedExtension(std::move(expectedExtension))
+{
+    try
+    {
+        IFile::checkFile();
+    }
+    catch (std::exception const &error)
```

Comment:
Perhaps add a comment such as `// We do not handle any exceptions, but re-throw everything instead` - or just drop the `try / catch`?